File: src/Web/Data/Match.cs
namespace RacketReel.Web.Data;

public class Match
{
    public int Id { get; private set; }
    public DateTime CreatedDateTime { get; private set; }
    public IEnumerable<string> Players { get; private set; }
    public string ServingFirst { get; private set; }
    public int Sets { get; private set; }
    public string SetType { get; private set; }
    public string FinalSetType { get; private set; }
    public IEnumerable<State> States { get; private set; }

    public Match(int id, DateTime createdDateTime, IEnumerable<string> players, string servingFirst, int sets, string setType, string finalSetType, IEnumerable<State> states)
    {
        Id = id;
        CreatedDateTime = createdDateTime;
        Players = players;
        ServingFirst = servingFirst;
        Sets = sets;
        SetType = setType;
        FinalSetType = finalSetType;
        States = states;
    }
}

File: src/racketreel_mobile/lib/matches/match.dart
import 'dart:ffi';

import 'package:racket_reel_mobile/matches/match_state.dart';
import 'package:racket_reel_mobile/matches/set_type.dart';

class Match {
  int id;
  DateTime createdDateTime;
  List<String> players;
  String servingFirst;
  int sets;
  SetType setType;
  SetType finalSetType;
  List<MatchState> states;

  Match(
    this.id,
    this.createdDateTime,
    this.players,
    this.servingFirst,
    this.sets,
    this.setType,
    this.finalSetType,
    this.states,
  );
}
